core/app/models/spree/promotion.rb

```diff
@@ -163,6 +164,12 @@ def eligible_rules(promotable, options = {})
         end
         specific_rules
       else
+        Spree::Deprecation.warn(
+        <<~WARN
+          Your promotion has a match_policy of 'any'.
```

Comment:
Maybe we can print the name/id of the promotion here to help people address the issue?

Reply:
Yes, that's easily possible!